examples/seismic/acoustic/acoustic_example.py

```diff
@@ -20,67 +20,52 @@ def smooth10(vel, shape):
     return out
 
 
-# Set up the source as Ricker wavelet for f0
-def source(t, f0):
-    r = (np.pi * f0 * (t - 1./f0))
-
-    return (1-2.*r**2)*np.exp(-r**2)
-
-
-def setup(dimensions=(50, 50, 50), spacing=(15.0, 15.0, 15.0), tn=500.,
-          time_order=2, space_order=4, nbpml=10, **kwargs):
+def acoustic_setup(dimensions=(50, 50, 50), spacing=(15.0, 15.0, 15.0),
+                   tn=500., time_order=2, space_order=4, nbpml=10, **kwargs):
 
     ndim = len(dimensions)
+    nrec = dimensions[0]
     origin = tuple([0.]*ndim)
     spacing = spacing[:ndim]
+
     # Velocity model, two layers
     true_vp = np.ones(dimensions) + .5
     true_vp[..., int(dimensions[-1] / 3):dimensions[-1]] = 2.5
-    # Source location
-    location = np.zeros((1, ndim), dtype=np.float32)
-    location[0, :-1] = [origin[i] + dimensions[i] * spacing[i] * .5
-                        for i in range(ndim-1)]
-    location[0, -1] = origin[-1] + 2 * spacing[-1]
-    # Receivers locations
-    receiver_coords = np.zeros((dimensions[0], ndim), dtype=np.float32)
-    receiver_coords[:, 0] = np.linspace(0, origin[0] +
-                                        (dimensions[0]-1) * spacing[0],
-                                        num=dimensions[0])
-    receiver_coords[:, 1:] = location[0, 1:]
 
     # Define seismic data
     model = Model(origin, spacing, dimensions, true_vp, nbpml=int(nbpml))
 
-    f0 = .010
+    # Derive timestepping from model spacing
     dt = model.critical_dt
     if time_order == 4:
         dt *= 1.73
     t0 = 0.0
-    tn = tn
-    nt = int(1+(tn-t0)/dt)
+    nt = int(1 + (tn-t0) / dt)
+    time = np.linspace(t0, tn, nt)
 
-    # Set up the source as Ricker wavelet for f0
-    def source(t, f0):
-        r = (np.pi * f0 * (t - 1./f0))
-        return (1-2.*r**2)*np.exp(-r**2)
+    # Define source geometry (center of domain, just below surface)
+    src = RickerSource(name='src', ndim=ndim, f0=0.01, time=time)
+    src.coordinates.data[0, :] = np.array(model.domain_size) * .5
```

Comment:
Overwritten in the next line; first put everything in the center then adjust the dimension we care about.